File: PSet1/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

// function prototype
int get_height();
void print_map(int height);

int main(void)
{
    // get height
    int height;
    height = get_height();

    // print map
    print_map(height);
}

// get height
int get_height()
{
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);
    return height;
}

// print the map
void print_map(int height)
{
    for (int i = 1; i <= height; i++)
    {
        // print the left part
        for (int j = 1; j <= height; j++)
        {
            if (j <= height - i)
            {
                printf(" ");
            }
            else
            {
                printf("#");
            }
        }

        // print two whitespaces
        printf("  ");

        // print the right part
        for (int j = 1; j <= i; j++)
        {
            printf("#");
        }
        printf("\n");
    }

}

File: Pset6/sentimental-mario-more/mario.py
# import cs50 module
from cs50 import get_int


def main():
    # get user's input
    height = get_height()

    for i in range(height):
        # print the left part
        print(" " * (height - i - 1), end="")
        print("#" * (i + 1), end="  ")

        # print the right part
        print("#" * (i + 1))

        # combine lines above into 1 line
        # print(" " * (height - i - 1) + "#" * (i + 1) + "  " + "#" * (i + 1))
        # print(" " * (height - i - 1), "#" * (i + 1), "  ", "#" * (i + 1), sep="")


def get_height():
    # do-while loop in Python
    while True:
        height = get_int("Height: ")
        if 1 <= height <= 8:
            return height


if __name__ == '__main__':
    main()